### ISSUE TITLE
[Bug]: Netherite Sluice with Fortune Upgrades lower output

### ISSUE THREAD

**Issue by Bauws** [FTBTeam/FTB-Mods-Issues#808]
### Mod

FTB Sluice

### Mod version

ftb-sluice-forge-1.0.11-build.47

### Forge / Fabric version

Forge 1.16.5 36.2.39

### Modpack & version

FTB Oceanblock 1.15.1

### What issue are you having?

Hey guys,

Setup:

- Netherite Sluice + Diamond Mesh
- Any infinite water source
- Creative energy cell
- Couple of Gravel stacks
- Fortune upgrades

Description:

Chance for iron chunk with the setup above is 69%. Without any upgrades the average drop was between 28-35 (which is completly fine). 

If you add 18x fortune upgrades the iron chunk drop rate per stack goes down to 13 on average. The expected behaviour would be more iron chunk drops with the fortune upgrades but it actually gets less.

I tested it with no upgrades, 9 and 18. No upgrades the drops are around 30 with any upgrades it goes down to 13. How is that possible? 1 upgrade should give 3% more chance and not less.

Best regards,
Bauws

### Steps to reproduce

1. Look at the description above please

**Comment by Bauws** [comment]
I think I have found the reason that the drops of iron chunks are lowered when using the fortune upgrade.
Every gravel that gets processed has 3 rolls. By increasing the chance for diamonds, emeralds and so on you increase the chance of getting those in one of the 3 rolls.

That is very useful for items that have a small loot pool but 3 rolls on gravel that has a total of 21 possible is just making it kinda tough to be worth it. Solution for that would be 1 sluice for iron and so the common ones and another with fortune upgrades for the more rare materials. 

I guess you put the rolls to 3 to not make it to powerful? I imagine giving each drop a roll would be to overpowered?

**Comment by TTblKK** [comment]
This also applies to different meshes. For example, when I am using diamond mesh, i recieve less iron chunks then when I am using iron mesh

**Comment by MichaelHillcox** [comment]
Hey! I know this is a super long time later but I've just had the time to dig into this with the original pack developer and yes! We both agree this is an issue but the solution here would break early game balance...

As this has been fixed in OceanBlock 2 (Stuff & Things), I'm going to opt to not push a fix for this. If the pack was newer and hadn't already had a V2, I may have worked something out but for most people, simply omitting the luck upgrades, at the point you have a sluice that can use them, and the cost of making them, the difference between a .2 chance and a .6 chance isn't actually that large...

tl;dr: I'd love to fix it but I'm to concerned for the balance. Effectively the mod only picks the first X amount of items before returning. This was an implementation issue as the spec was meant to be that the entire loot table was rolled X times, not capped at X.

Hope this helps, happy to hear any feedback on this! 

Gunna close for now but as I've replied in here, I'll get pings for any updates to this issue <3

**Comment by Bauws** [comment]
Greetings Michael,

thank you for taking a look at it. You got a big project it is normal that it takes a while to take a look at things, so dont worry :) 

I totally understand your points, making this to overpowered would break the game. I think we even had that in a different version with exnihilio where the meshes just dropped insane amounts.

I wish you a wonderful week and maximum success :)

Best regards,
Bauws
